doc/src/devdocs/types.md

```diff
@@ -455,129 +335,34 @@ breakpoints in other functions.
 As a warm-up, try the following:
 
 ```julia
-myintersect(Tuple{Integer,Float64}, Tuple{Int,Real})
-```
-
-Set a breakpoint in `intersect_tuple` and continue until it enters this function.  You should
-be able to see something like this:
-
-```
-Breakpoint 2, intersect_tuple (a=0x7ffdf7409150, b=0x7ffdf74091b0, penv=0x7fffffffcc90, eqc=0x7fffffffcc70, var=covariant) at jltypes.c:405
-405     {
-(gdb) call jl_(a)
-Tuple{Integer, Float64}
-(gdb) call jl_(b)
-Tuple{Int64, Real}
-```
-
-The `var` argument is either `covariant` or `invariant`, the latter being used if you're matching
-the type parameters of `Array{T1}` against `Array{T2}`.  The other two inputs to this function
-(`penv` and `eqc`) may be currently mysterious, but we'll discuss them in a moment.  For now,
-step through the code until you get into the loop over the different entries in the tuple types
-`a` and `b`.  The key call is:
-
-```julia
-ce = jl_type_intersect(ae,be,penv,eqc,var);
+mysubtype(Tuple{Int,Float64}, Tuple{Integer,Real})
 ```
 
-which, if you examine `ae`, `be`, and `ce`, you'll see is just type intersection performed on
-these entries.
-
 We can make it more interesting by trying a more complex case:
 
 ```julia
-julia> T = TypeVar(:T, true)
-T
-
-julia> myintersect(Tuple{Array{T}, T}, Tuple{Array{Int,2}, Int8})
-
-Breakpoint 1, jl_breakpoint (v=0x7ffdf35e8010) at builtins.c:1559
-1559    {
-(gdb) b intersect_tuple
-Breakpoint 3 at 0x7ffff6dcb07d: file jltypes.c, line 405.
-(gdb) c
-Continuing.
-
-Breakpoint 3, intersect_tuple (a=0x7ffdf74d7a90, b=0x7ffdf74d7af0, penv=0x7fffffffcc90, eqc=0x7fffffffcc70, var=covariant) at jltypes.c:405
-405     {
-(gdb) call jl_(a)
-Tuple{Array{＃T<:Any, N<:Any}, ＃T<:Any}
-(gdb) call jl_(b)
-Tuple{Array{Int64, 2}, Int8}
-```
-
-Let's watch how this bound `TypeVar` gets handled.  To follow this, you'll need to examine the
-variables `penv` and `eqc`, which are defined as:
-
-```
-typedef struct {
-    jl_value_t **data;
-    size_t n;
-    jl_svec_t *tvars;
-} cenv_t;
-```
-
-These start out empty (with `penv->n == eqc->n == 0`).  Once we get into the loop and make the
-first call to `jl_type_intersect`, `eqc` (which stands for "equality constraints") has the following
-value:
-
+mysubtype(Tuple{Array{Int,2}, Int8}, Tuple{Array{T}, T} where T)
 ```
-(gdb) p eqc->n
-$4 = 2
-(gdb) call jl_(eqc->data[0])
-＃T<:Any
-(gdb) call jl_(eqc->data[1])
-Int64
-```
-
-This is just a `var`, `value` list of pairs, indicating that `T` now has the value `Int64`.  If
-you now allow `intersect_tuple` to finish and keep progressing, you'll eventually get to `type_intersection_matching`.
- This function contains a call to `solve_tvar_constraints`.  Roughly speaking, `eqc` defines
-`T = Int64`, but `env` defines it as `Int8`; this conflict is detected in `solve_tvar_constraints`
-and the resulting return is `jl_bottom_type`, aka `Union{}`.
```

Comment:
I know, which is why I wanted to now read about all of the new data structures and algorithms